tests/test_exe.py

```diff
@@ -913,3 +913,101 @@ def test_pbs_provider_pre_releases(tmp_path: Path, science_exe: Path) -> None:
         "3.15.0a1"
         == subprocess.run(args=[scie], stdout=subprocess.PIPE, text=True, check=True).stdout.strip()
     )
+
+
+def test_pbs_provider_freethreaded_builds(tmp_path: Path, science_exe: Path) -> None:
+    dest = tmp_path / "dest"
+    chroot = tmp_path / "chroot"
+    chroot.mkdir(parents=True, exist_ok=True)
+
+    exe = tmp_path / "exe"
+    exe.write_text(
+        dedent(
+            """\
+            import platform
+            import sysconfig
+
+
+            if __name__ == "__main__":
+                print(platform.python_version())
+                print(sysconfig.get_config_var("Py_GIL_DISABLED"))
+            """
+        )
+    )
+
+    match Platform.current():
+        case Platform.Linux_x86_64 if LibC.current() is LibC.GLIBC:
+            flavor = "freethreaded+pgo+lto-full"
+        case Platform.Macos_aarch64 | Platform.Macos_x86_64:
+            flavor = "freethreaded+pgo+lto-full"
+        case _:
+            flavor = "freethreaded+pgo-full"
```

Comment:
N.B.: The flavor for free-threaded builds is platform specific currently over in PBS.